src/Streamly/Internal/Data/Fold/Types.hs

```diff
@@ -403,25 +607,72 @@ ltake n (Fold step initial done) = Fold step' initial' done'
         if i < n
         then do
             res <- step r a
-            return $ Tuple' (i + 1) res
-        else return $ Tuple' i r
+            case res of
+                Partial s -> return $ Partial $ Tuple' (i + 1) s
+                Done b -> return $ Done b
```

Comment:
We have a problem with `ltake 0`, we will lose the current input. We should use the `Reject`/`Done1` constructor in that case.

Reply:
We should use the `take` parser code here and the `take` parser can be removed once we have terminating folds committed. Same is true for takeWhile/all/any.